contracts/Rollup.sol

```diff
@@ -283,7 +278,9 @@ contract Rollup is RollupHelpers {
             )
         });
         batches.push(newBatch);
-        stake(batches.length - 1);
+        uint256 batchID = batches.length - 1;
+        stake(batchID);
+        logger.logNewBatch(msg.sender, batchID, batchType);
```

Comment:
I wonder if not creating a batchID variable is cheaper.

Reply:
Sounds about right